File: LeetCode/Challenges/0001 Two Sum.cs
namespace LeetCode.Challenges._0001TwoSum;

public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        for (int i = 0; i < (nums.Length - 1); i++) {
            for (int j = i + 1; j < nums.Length; j++) {
                if (nums[i] + nums[j] == target) {
                    return new int[2] { i, j };
                }
            }
        }
        return new int[0];
    }
}

File: go/challenges/0001_two_sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	for i, num := range nums {
		for j, num2 := range nums[i+1:] {
			j += i + 1
			if num+num2 == target {
				return []int{i, j}
			}
		}
	}
	return nil
}

func main() {
	input := []int{2, 7, 11, 15}
	target := 9
	expectedResult := []int{0, 1}

	result := twoSum(input, target)
	fmt.Println("Result:", result)
	fmt.Println("Expected:", expectedResult)
}
